core/src/main/scala/kafka/server/LogOffsetMetadata.scala

```diff
@@ -51,25 +51,15 @@ case class LogOffsetMetadata(messageOffset: Long,
   }
 
   // check if this offset is on the same segment with the given offset
-  def offsetOnSameSegment(that: LogOffsetMetadata): Boolean = {
+  def onSameSegment(that: LogOffsetMetadata): Boolean = {
     if (messageOffsetOnly())
       throw new KafkaException(s"$this cannot compare its segment info with $that since it only has message offset info")
 
     this.segmentBaseOffset == that.segmentBaseOffset
   }
 
-  // check if this offset is before the given offset. We need to compare both message offset and segment base offset.
-  def precedes(that: LogOffsetMetadata): Boolean = {
-    this.messageOffset < that.messageOffset || {
-      if (this.messageOffsetOnly() && that.messageOffsetOnly())
-        false
-      else if (!this.messageOffsetOnly() && !that.messageOffsetOnly())
-        this.segmentBaseOffset < that.segmentBaseOffset
-      else
-        throw new KafkaException(s"Cannot compare $this with $that as one has segment base offsets and the other does not.")
-    }
-  }
-
+  // check if the message offset is before the given offset.
+  def precedes(that: LogOffsetMetadata): Boolean = this.messageOffset < that.messageOffset
```

Comment:
I agree and I'd probably go with option 1, but either is fine with me too. The fact that we've had at least 2 separate commits where the meaning of `precedes` wasn't well understood tells me that it's not a name we want to keep, even though it reads much more nicely. :)